### PR TITLE
Ignore empty token cookies

### PR DESCRIPTION
**Pull request by ajacksified**
Instead of trying to refresh a token when a tokenExpires exists but a
token doesn't, nuke the invalid cookies.

:eyeglasses: @schwers @umbrae

### CODE REVIEW

**Comment by ajacksified** [comment]
Fixes #430

File: src/server/index.jsx [hunk 1]
```diff
@@ -443,6 +444,11 @@ class Server {
 
           return;
         }
+      // Sometimes, there's an empty token cookie.
```
Comment (umbrae) [inline src/server/index.jsx]: I wonder if this should be a bit more clear of the particular issue we're seeing, or the edge case we're trying to handle? This comment might make it sound like this is a desired/expected result, but it's not, so we might want to say something explicit about this being error handling for malformed cookies.

**Comment by umbrae** [comment]
:nail_care: code looks good, just one comment thought

File: src/server/index.jsx [hunk 2]
```diff
@@ -420,16 +420,17 @@ class Server {
...
+      const rToken = this.cookies.get('refreshToken');
 
-      if (now > expires) {
-        var rToken = this.cookies.get('refreshToken');
+      const expires = new Date(cookieExpires);
 
+      if (cookieToken && rToken && now > expires) {
         try {
-          var token = yield app.refreshToken(this, rToken);
+          let token = yield app.refreshToken(this, rToken);
```
Comment (schwers) [inline src/server/index.jsx]: This should probably be `const`

**Comment by schwers** [comment]
:nail_care: small nit, but looks good

**Comment by ajacksified** [comment]
:haircut:

**Comment by spladug** [comment]
General question: why two cookies instead of one if their content is conjoined?

**Comment by umbrae** [comment]
:fish: on this fix - we probably could join these cookies, but that would be a larger thing. (I think we got bit by this same thing with `loid` and `loidcreated` :frowning:)

**Comment by ajacksified** [comment]
Leftover implementation detail from before we had refresh tokens
implemented. Another PR should set a matching expiry on the token instead
of keeping a separate tokenExpires cookie, and remove calls to tokenExpires.

**Comment by ajacksified** [comment]
Err, thinking about it again, I think it was also a cheap way of getting
expiry to know when to refresh, instead of reading the expires from the
cookie. No reason it all couldn't be in one base64'd json blob or something.

**Pull request merged**